src/bt_peer_connection.cpp

```diff
@@ -3136,14 +3397,37 @@ namespace {
 				std::copy(recv_buffer.begin() + 8, recv_buffer.begin() + 28
 					, info_hash.data());
 
-				attach_to_torrent(info_hash);
+				attach_to_torrent(info_hash_t(info_hash));
 				if (is_disconnecting()) return;
+
+				// this must go after the connection is attached to a torrent because that is what
+				// adds the peer info for incoming connections
+				if (recv_buffer[7] & 0x10)
+				{
+					if (!associated_torrent().lock()->info_hash().has_v2())
+					{
+						// the peer claims to support the v2 protocol with a non-v2 torrent
```

Comment:
is it not possible that the "v2" aspect of a torrent is stripped away via a magnet link? and that this could happen legitimately (i.e. non-maliciously)

Reply:
Good catch. This check should only be done if the torrent has valid metadata. I'll fix this.